guppylang/prelude/builtins.py

```diff
@@ -75,6 +75,9 @@ def __and__(self: bool, other: bool) -> bool: ...
     @guppy.custom(builtins, NoopCompiler())
     def __bool__(self: bool) -> bool: ...
 
+    @guppy.hugr_op(builtins, logic_op("Eq", [tys.TypeArg(tys.BoundedNatArg(n=2))]))
```

Comment:
shouldn't n=1?

Reply:
The argument here is to the template for boolean ops, the 2 is specifying how many inputs the function has, rather than the bound of the input Ints